lib/Normalize_std_ast.ml

```diff
@@ -216,61 +212,116 @@ let make_mapper conf ~ignore_doc_comments ~erase_jane_syntax =
+             constraining based on modes, remove the node entirely instead of
+             just making the modes list empty *)
+          m.expr m exp1
+      | Pexp_constraint (e, Some {ptyp_desc= Ptyp_poly ([], _t); _}, []) ->
+          m.expr m e
+      | Pexp_sequence
+          ( exp1
+          , { pexp_desc= Pexp_sequence (exp2, exp3)
+            ; pexp_loc= loc2
+            ; pexp_attributes= attrs2
+            ; _ } ) ->
+          m.expr m
+            (Exp.sequence ~loc:loc1 ~attrs:attrs1
+               (Exp.sequence ~loc:loc2 ~attrs:attrs2 exp1 exp2)
+               exp3 )
+      | Pexp_function
+          ( ps
+          , c
+          , Pfunction_body {pexp_desc= Pexp_function (ps', None, b'); _} ) ->
+          m.expr m {exp with pexp_desc= Pexp_function (ps @ ps', c, b')}
+      | Pexp_newtype (l, {pexp_desc= Pexp_constraint (exp1, Some ty, []); _})
+        ->
+          (* It's strange that this creates an illegal [Pexp_function] node
+             (with a single [Pparam_newtype] parameter and a
+             [Pfunction_body]), but it's necessary to normalize against a
+             strange ast change in [source.ml]. This can probably be deleted
+             when we merge with 5.2 ocamlformat *)
+          m.expr m
+            { exp with
+              pexp_desc=
+                Pexp_function
+                  ( [ { pparam_loc= l.loc
+                      ; pparam_desc= Pparam_newtype (l, None) } ]
+                  , Some
+                      {mode_annotations= []; type_constraint= Pconstraint ty}
+                  , Pfunction_body exp1 ) }
+      | Pexp_function
+          ( ps
+          , c
+          , Pfunction_body {pexp_desc= Pexp_constraint (exp1, Some ty, []); _}
+          ) ->
+          let c =
+            match c with
+            | Some c -> Some {c with type_constraint= Pconstraint ty}
+            | None ->
+                Some {mode_annotations= []; type_constraint= Pconstraint ty}
           in
-          {exp with pexp_desc}
-        in
-        m.expr m expression
-    | Pexp_extension ({txt= "src_pos"; loc}, _) when erase_jane_syntax ->
-        m.expr m (dummy_position ~loc)
-    | Pexp_fun _ | Pexp_function _ | Pexp_newtype _ ->
-        (* CR jane-syntax: This just ignores N_ary functions, and can be
-           removed when ocamlformat stops messing with them *)
-        let attrs1 =
-          List.filter ~f:(fun a -> not (is_builtin_jane_syntax a)) attrs1
-        in
-        Ast_mapper.default_mapper.expr m {exp with pexp_attributes= attrs1}
-    | _ -> Ast_mapper.default_mapper.expr m exp
+          m.expr m
+            {exp with pexp_desc= Pexp_function (ps, c, Pfunction_body exp1)}
+      | Pexp_function (ps, c, b) when erase_jane_syntax ->
+          let ps =
+            List.map ps ~f:(fun param ->
+                match param.pparam_desc with
+                | Pparam_newtype (x, Some _) ->
+                    {param with pparam_desc= Pparam_newtype (x, None)}
```

Comment:
Why throw away the jkind here?  I don't see similar logic in the old version.

Reply:
This is `when erase_jane_syntax`, where presumably we want to remove jkind annotations. Before this was just happening by removing the jane syntax attributes (see the `CR jane-syntax` at the top of this function)